### PR TITLE
Fix spring-web test failure

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

**Review by geoand — APPROVED** [review]

**Comment by geoand** [comment]
Was this intended to fix the `SpringControllerTest.testInvalidJsonInputAndResult` failure?

Asking because it was fixed with https://github.com/quarkusio/quarkus/pull/20271

**Comment by stuartwdouglas** [comment]
I did not notice that, but do we really want to be adding a resteasy-classic SPI artifact to a resteasy-reactive module?

**Comment by geoand** [comment]
Damn it, you are right... 

It's just too hard to review all the RESTEasy stuff in the Github UI... It's just too hard to get mixed up regarding the module that is affected... Now I know to pay much more attention.

Would you like to update this PR with a commit that reverts https://github.com/quarkusio/quarkus/pull/20271?

**Comment by geoand** [comment]
Otherwise, I can do it. I just don't want to have the build broken again before the proper fix is in

**Comment by yrodiere** [comment]
> I did not notice that, but do we really want to be adding a resteasy-classic SPI artifact to a resteasy-reactive module?

I though that was fine, because the SPI does not depend on resteasy-classic, and it also is very specific to Quarkus. So AFAICS, we could use the same SPI for both.

**Comment by geoand** [comment]
> I though that was fine, because the SPI does not depend on resteasy-classic, and it also is very specific to Quarkus. So AFAICS, we could use the same SPI for both.

Technically yeah, it's not a problem. But we really want to the two to be totally separate.

**Comment by stuartwdouglas** [comment]
Should we just have a common SPI module for stuff like this (where it does not actually depend on anything specific to one or the other)?

**Comment by geoand** [comment]
Yeah, for things that are common, that makes sense.

I'll looke into it after `2.3.0.CR1` is out.

**Comment by stuartwdouglas** [comment]
I have added an SPI module just with this class as a starting point.

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java [hunk 1]
```diff
@@ -432,15 +434,15 @@ private static void contributeMethodsWithInheritedValidation(
...
         Map<DotName, Set<SimpleMethodSignatureKey>> jaxRsMethods = new HashMap<>();
 
         Collection<DotName> jaxRsMethodDefiningAnnotations = new ArrayList<>(
                 ResteasyDotNames.JAXRS_METHOD_ANNOTATIONS.size() + additionalJaxRsResourceMethodAnnotations.size());
         jaxRsMethodDefiningAnnotations.addAll(ResteasyDotNames.JAXRS_METHOD_ANNOTATIONS);
-        for (AdditionalJaxRsResourceMethodAnnotationsBuildItem additionalJaxRsResourceMethodAnnotation : additionalJaxRsResourceMethodAnnotations) {
-            jaxRsMethodDefiningAnnotations.addAll(additionalJaxRsResourceMethodAnnotation.getAnnotationClasses());
+        for (DotName additionalJaxRsResourceMethodAnnotation : additionalJaxRsResourceMethodAnnotations) {
+            jaxRsMethodDefiningAnnotations.add(additionalJaxRsResourceMethodAnnotation);
         }
```
Comment (yrodiere) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java:446]: I know it's a detail, but if you're going to amend the commit anyway...

```suggestion
        jaxRsMethodDefiningAnnotations.addAll(additionalJaxRsResourceMethodAnnotations);
```

File: extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java [hunk 2]
```diff
@@ -314,7 +314,9 @@ public void build(HibernateValidatorRecorder recorder, RecorderContext recorderC
         // JAX-RS methods are handled differently by the transformer so those need to be gathered here.
         // Note: The focus only on methods is basically an incomplete solution, since there could also be
         // class-level JAX-RS annotations but currently the transformer only looks at methods.
-        Map<DotName, Set<SimpleMethodSignatureKey>> jaxRsMethods = gatherJaxRsMethods(additionalJaxRsResourceMethodAnnotations,
+        Set<DotName> additional = new HashSet<>();
+        additionalJaxRsResourceMethodAnnotations.forEach((s) -> additional.addAll(s.getAnnotationClasses()));
```
Comment (yrodiere) [inline extensions/hibernate-validator/deployment/src/main/java/io/quarkus/hibernate/validator/deployment/HibernateValidatorProcessor.java:318]: I think you forgot to update the import of `AdditionalJaxRsResourceMethodAnnotationsBuildItem` from the old SPI (it should import the one from the new SPI).

Maybe you could also remove that class from the old SPI, just to be sure it doesn't get used again by mistake?

**Review by yrodiere — CHANGES_REQUESTED** [review]
I think the reason tests are still failing is an import that wasn't updated; see below.

**Review by yrodiere — APPROVED** [review]
Damn that was fast.

**Review by geoand — APPROVED** [review]

**Pull request merged**